core/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -113,27 +120,27 @@ public void deprecated(String msg, Object... params) {
      * Logs a deprecated message to the deprecation log, as well as to the local {@link ThreadContext}.
      *
      * @param threadContexts The node's {@link ThreadContext} (outside of concurrent tests, this should only ever have one context).
-     * @param msg The deprecation message.
+     * @param message The deprecation message.
      * @param params The parameters used to fill in the message, if any exist.
      */
     @SuppressLoggerChecks(reason = "safely delegates to logger")
-    void deprecated(Set<ThreadContext> threadContexts, String msg, Object... params) {
+    void deprecated(Set<ThreadContext> threadContexts, String message, Object... params) {
         Iterator<ThreadContext> iterator = threadContexts.iterator();
 
         if (iterator.hasNext()) {
-            final String formattedMsg = LoggerMessageFormat.format(msg, params);
+            final String formattedMessage = LoggerMessageFormat.format(message, params);
 
             while (iterator.hasNext()) {
                 try {
-                    iterator.next().addResponseHeader(DEPRECATION_HEADER, formattedMsg);
+                    iterator.next().addResponseHeader(DEPRECATION_HEADER, formattedMessage);
                 } catch (IllegalStateException e) {
                     // ignored; it should be removed shortly
                 }
             }
 
-            logger.debug(formattedMsg);
+            logger.warn(formattedMessage);
```

Comment:
After #20254, we have consensus to do this out-of-the-box by default, and I will configure the deprecation logger to be size-limited.